packages/liferay-js-insights/insights/dependencies.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/**
+ * Extracts import declarations from a given AST applying the following transforms:
+ * - Relative imports are discarded
+ * - Imports are grouped per origin into:
+ * 	- clay3
+ *  - react
+ *  - js
+ *  - metal
+ *  - clay2
+ *  - others
+ */
+function extractImports(ast) {
+	const EMPTY = [[], [], [], [], [], []];
```

Comment:
This is a bit fragile because you have to keep it in sync with the list below. You could instead create it based on the `length` of `SOURCE_GROUPS`.

```javascript
const EMPTY = Array.from(new Array(SOURCE_GROUPS.length), () => []);
```